admin/spam/views.py

```
@@ -101,24 +98,38 @@ def post(self, request, *args, **kwargs):
         # return super(SpamDetail, self).post(request, *args, **kwargs)
 
     def get_context_data(self, **kwargs):
-        self.item = Comment.load(self.kwargs.get('spam_id', None))
+        item = Comment.load(self.kwargs.get('spam_id', None))
         kwargs = super(SpamDetail, self).get_context_data(**kwargs)
         kwargs.setdefault('page_number', self.request.GET.get('page', 1))
-        kwargs.setdefault('comment', serialize_comment(self.item))
+        kwargs.setdefault('comment', serialize_comment(item))
         kwargs.setdefault('status', self.request.GET.get('status', u'1'))
         return kwargs
 
     def form_valid(self, form):
+        spam_id = self.kwargs.get('spam_id', 'no find')
```

This should probably raise some exception if `spam_id` is missing